File: backend/subscription-service/src/main/java/by/belpost/subscription_service/repository/PublicationRepository.java
package by.belpost.subscription_service.repository;

import by.belpost.subscription_service.entity.Publication;
import by.belpost.subscription_service.enums.PublicationType;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface PublicationRepository extends JpaRepository<Publication, Long> {

    List<Publication> findByType(PublicationType type);

    List<Publication> findByCategoriesId(Long categoryId);

    List<Publication> findByTypeAndCategoriesId(PublicationType type, Long categoryId);
}

File: android/BelpostSubscription/app/src/main/java/com/belpost/subscription/data/repository/PublicationRepository.kt
package com.belpost.subscription.data.repository

import com.belpost.subscription.data.api.ApiService
import com.belpost.subscription.data.api.models.PublicationDto

class PublicationRepository(
    private val apiService: ApiService
) {

    suspend fun getPublications(
        type: String? = null,
        categoryId: Long? = null
    ): List<PublicationDto> {
        return apiService.getPublications(type, categoryId)
    }
}

